lambda_function.py

```diff
@@ -0,0 +1,61 @@
+import os
+import json
+import urllib3
+import boto3
+from urllib3.exceptions import ConnectTimeoutError
+from boto3 import resource
+from botocore.exceptions import ClientError
+from boto3.dynamodb.conditions import Key, Attr
+
+convert_value_to_bool = {"true": True, "false": False}
+attribute_to_add = "ingested_CC"
+
+
+def get_items_with_id(client, table_name, primary_key, primary_key_value):
+    response = client.query(
+        TableName=table_name,
+        KeyConditionExpression=f"{primary_key} = :value",
+        ExpressionAttributeValues={":value": {"S": primary_key_value}}
+    )
+    items = response["Items"]
+
+    return items
+
+
+def add_true_to_ingest_cc_attribute(client, table_name, primary_key, sort_key, items_with_id):
+    for item_with_id in items_with_id:
+        attribute_type_and_value = item_with_id.get(attribute_to_add, {"S": "false"})
+        attribute_value = attribute_type_and_value["S"]
+        attribute_value_is_true = convert_value_to_bool.get(attribute_value, False)
+
+        if not attribute_value_is_true:
+            primary_key_type_and_value = item_with_id[primary_key]
+            sort_key_value_type_and_value = item_with_id[sort_key]
+
+            client.update_item(
+                TableName=table_name,
+                Key={primary_key: primary_key_type_and_value, sort_key: sort_key_value_type_and_value},
+                UpdateExpression=f"SET {attribute_to_add} = :ingestedCCValue",
+                ExpressionAttributeValues={":ingestedCCValue": {"S": "true"}},
+                ConditionExpression=f"attribute_exists({primary_key})"
+            )
```

Comment:
Could we simplify this by using the Boolean type in Dynamo?

Reply:
I'm sure there was a reason we didn't although it's not coming to mind now. `ingested_PS` is a string so maybe leave this as a string as well for consistency.

Reply:
@MancunianSam, is this the reason

> Because this isn't an RDS, we can't have a boolean type for what is a
> boolean column so this converts between the boolean in the case class
> and the string in the column.

Source: [you]((https://github.com/nationalarchives/dr2-dynamo-formatters/pull/27))

Reply:
:smile: I guess that was on the DynamoFormatters project. It doesn't explain why we can't have a boolean type in Dynamo but I'm happy to blame Dynamo and leave it as a string.